std/shared/src/main/scala/cats/effect/std/Hotswap.scala

```
@@ -64,11 +64,20 @@ sealed trait Hotswap[F[_], R] {
    * is not used thereafter. Failure to do so may result in an error on the _consumer_ side. In
    * any case, no resources will be leaked.
    *
+   * For safer access to the current resource see [[get]], which guarantees that it will not be
+   * released while it is being used.
+   *
    * If [[swap]] is called after the lifetime of the [[Hotswap]] is over, it will raise an
    * error, but will ensure that all resources are finalized before returning.
    */
   def swap(next: Resource[F, R]): F[R]
 
+  /**
+   * Gets the current resource, if it exists. The returned resource is guaranteed to be
+   * available for the duration of the returned resource.
+   */
+  def get: Resource[F, Option[R]]
```

Maybe a little bit annoying UX that this is an `Option[R]`, alternatively we could raise an error. But the user can decide to do that too.

---

Bikeshed the name. `access` a la `HotswapRef` ?